ykllvmwrap/src/jitmodbuilder.cc

```diff
@@ -165,6 +165,26 @@ class JITModBuilder {
     }
   }
 
+  void handleReturnInst(Instruction *I) {
+    ResumeAfter = InlinedCalls.back();
+    InlinedCalls.pop_back();
+    if (call_stack > 0) {
+      call_stack -= 1;
+      if (call_stack == 0) {
+        ResumeAfter = NoInlineFunc;
```

Comment:
Is it just me, or does this assign to a variable, then the function returns, so the variable assignment can't have any effect?

Reply:
It's a field.

[One of the tasks on my list is to minimise the number of fields in the class, but that will be a separate PR]

Reply:
Can we use `this` or `self` or whatever for these? IMHO, the "implicit field thing" is a horrible language feature!

Reply:
You can use the `this` pointer, but it's only really idiomatic to use it when disambiguating a local variable of the same name.

Reply:
Java has the same expectation... and it seems stupid to me. It reduces one's ability to reason about code locally (as I've just shown), particularly in diffs. We don't have to change all the existing code, but IMHO we should consider using `this` for new stuff. Anyway, not an issue here and now.

Reply:
I agree that it's not nice, but since we've adopted LLVM's coding conventions for our C++ code, I think the right thing to do is follow the precedent.

[It's not *that* bad. If you see a variable undefined locally, it's in another scope]

Reply:
> [It's not that bad. If you see a variable undefined locally, it's in another scope]

It really is bad, because now I have to *remember* the entire set of local variables to know if something's a variable or a field -- two things that have very different semantics. For diffs this means I have to look at the entire function, otherwise (as in this case) I will get completely the wrong impression. Anyway, I won't insist upon this, and it's not a concern for this PR.

Reply:
To be clear, I totally agree. But it's no worse than global variables. They too force you to look around more :P